Code/GraphMol/SubstanceGroup.cpp

```diff
@@ -294,6 +295,74 @@ unsigned int addSubstanceGroup(ROMol &mol, SubstanceGroup sgroup) {
   return id;
 }
 
+namespace {
+bool includesBond(SubstanceGroup &sg, unsigned int idx) {
+  return sg.includesBond(idx);
+}
+bool includesAtom(SubstanceGroup &sg, unsigned int idx) {
+  return sg.includesAtom(idx);
+}
+void removedBond(SubstanceGroup &sg, unsigned int idx) {
+  sg.adjustToRemovedBond(idx);
+}
+void removedAtom(SubstanceGroup &sg, unsigned int idx) {
+  sg.adjustToRemovedAtom(idx);
+}
+
+template <bool INCLUDES_METHOD(SubstanceGroup &, unsigned int),
+          void ADJUST_METHOD(SubstanceGroup &, unsigned int)>
+void removeSubstanceGroupsReferencing(RWMol &mol, unsigned int idx) {
+  auto &sgs = getSubstanceGroups(mol);
+  if (!sgs.empty()) {
+    // first collect a vector the ones that should be removed
+    std::vector<unsigned> toRemove;
+    for (auto &&sg : sgs) {
+      if (INCLUDES_METHOD(sg, idx)) {
+        unsigned int index;
+        if (sg.getPropIfPresent("index", index)) {
+          toRemove.push_back(index);
+        }
+      }
```

Comment:
Do we have to care about SD indexes here? We could just capture vector indexes, and worry about SD index properties in the next loop, only if we ever hit any parent properties.

Reply:
yep, makes sense. Made that change too